packages/flutter/lib/src/material/dropdown.dart

```diff
@@ -1040,6 +1034,12 @@ class _DropdownButtonState<T> extends State<DropdownButton<T>> with WidgetsBindi
   Widget build(BuildContext context) {
     assert(debugCheckHasMaterial(context));
     assert(debugCheckHasMaterialLocalizations(context));
+    final Orientation newOrientation = MediaQuery.of(context).orientation;
+    _lastOrientation ??= newOrientation;
+    if (MediaQuery.of(context).orientation != _lastOrientation) {
```

Comment:
Could this be changed to the following instead: 
```suggestion
    if (newOrientation != _lastOrientation) {
```

If I understand `??=` correctly, this would help us avoid looking up `MediaQuery.of(context).orientation` twice